tests/backwards_compatibility_tests/compatibility_test_runner.py

```diff
@@ -212,35 +207,46 @@ def start_marqo_from_version_container(from_version: str, from_version_volume, e
-                                        This is constructed in build_push_image.yml workflow and will be the qualified image name with digest for an automatically triggered workflow.
-        env_vars (Optional[list]): A list of environment variables to set in the container. Defaults to None.
+        target_version (str): The target version of the Marqo container to start. This variable will contain 'to_version' in case of a backwards compatibility test
+                                    whereas it will contain both 'to_version' and 'from_version' (albeit in different method calls) in case of rollback tests.
+        source_version (str): The source version of the Marqo container. The source_marqo_version parameter is later used to determine how we transfer state.
+                                    This variable will contain 'from_version' in case of a backwards compatibility test whereas it will contain both 'from_version'
+                                    and 'to_version' (albeit in different method calls) in case of rollback tests.
+        source_volume (str): The volume to use for the container.
+        target_version_image_identifier (str): The unique identifier for a target_version image. It can be either be the fully qualified image name with the tag (ex: 424082663841.dkr.ecr.us-east-1.amazonaws.com/marqo-compatibility-tests:abcdefgh1234)
+                                    or the fully qualified image name with the digest (ex: 424082663841.dkr.ecr.us-east-1.amazonaws.com/marqo-compatibility-tests@sha256:1234567890abcdef).
+                                    This is constructed in build_push_image.yml workflow and will be the qualified image name with digest for an automatically triggered workflow.
+                                    Imp. Note: This parameter only needs to be passed in case we are trying to start a container that doesn't exist on dockerHub, essentially only pass this parameter if the source is ECR.
+        source (str): The source from which to pull the image. It can be either 'docker' for Docker Hub or 'ECR' for Amazon ECR. Not to be confused with source_version.
     """
-    source = "ECR" #Source of a to_version image will always be ECR because we build and push unpublished & to be tested images to ECR
+
     logger.debug(
-        f"Starting Marqo container with to_version: {to_version}, "
-        f"from_version: {from_version} "
-        f"from_version_volume: {from_version_volume}, to_version_identifier: {to_version_identifier}, source: {source}")
-    container_name = f"marqo-{to_version}"
-    to_version = semver.VersionInfo.parse(to_version)
-    from_version = semver.VersionInfo.parse(from_version)
+        f"Starting Marqo container with target version: {target_version}, "
+        f"source version: {source_version} "
```

Comment:
Are we being consistent with the terminology? We shouldn't use both source version and from version especially in input args and outputs